### PR TITLE
put seed nodes into separate file

### PR DESCRIPTION
**Pull request by crypto-ape**
Hey Monkeys!

This change isolates the hard-coded seed nodes into separate file.

Ape out!

### CODE REVIEW

**Comment by oxarbitrage** [comment]
- I think it can be a good idea to put this in a separated file but i will like opinions from others.

- I personally dont like the implementation with the `#include` in the middle of the code. Even if in this case it can be harmless, it just dont looks right. We can read the text file line by line and it should be a better approach IMHO.

- The location of this file should not be at root, maybe `libraries/egenesis` or `programs/witness_node` are better places.

**Comment by 01mk** [comment]
This is brilliant lazy programmer solution :D.

It would be much better to read it from some kind of config file.

**Comment by crypto-ape** [comment]
Yes, it was a brilliantly pragmatic solution.

- I agree that the file should be located more appropriately, 'libraries/egenesis' does not sound bad.
- Yes, the raw include is both elegant and ugly, not systematic. Reading a file line by line sounds right.

I will wait a while before refactoring it until a more profound consensus is reached.

**Comment by pmconrad** [comment]
>  the raw include is both elegant and ugly

Agree. I like it. :-)

I think you can get rid of the `CMakeLists` change if you `#include` with double quotes not angled brackets.

File: seed-nodes.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+"seed.bitsharesnodes.com:1776",      // wackou       (Netherlands)
+"seed.blocktrades.us:1776",          // BlockTrades  (USA)
+"seed.cubeconnex.com:1777",          // cube         (USA)
+"seed.roelandp.nl:1776",             // roelandp     (Canada)
+"seed04.bts-nodes.net:1776",         // Thom         (Australia)
+"seed05.bts-nodes.net:1776",         // Thom         (USA)
+"seed06.bts-nodes.net:1776",         // Thom         (USA)
+"seed07.bts-nodes.net:1776",         // Thom         (Singapore)
+"seed.bts.bangzi.info:55501",        // Bangzi       (Germany)
+"seeds.bitshares.eu:1776",           // pc           (http://seeds.quisquis.de/bitshares.html)
```
Comment (pmconrad) [inline seed-nodes.txt]: The last line shouldn't have a `,`

**Comment by pmconrad** [comment]
> It would be much better to read it from some kind of config file.

We *have* a config file. The list here is meant to provide a working hardcoded default.
IMO using a tool like `embed_genesis` for something that can be solved with a simple `#include` is overkill.

File: seed-nodes.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+"seed01.liondani.com:1776",          // liondani     (GERMANY)
+"104.236.144.84:1777",               // puppies      (USA)
```
Comment (pmconrad) [inline seed-nodes.txt]: Inactive, please remove

File: seed-nodes.txt [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+"128.199.143.47:2015",               // Harvey       (Singapore)
```
Comment (pmconrad) [inline seed-nodes.txt]: Inactive, please remove

File: seed-nodes.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+"209.105.239.13:1776",               // sahkan       (USA)
+"45.35.12.22:1776",                  // sahkan       (USA)
+"51.15.61.160:1776",                 // lafona       (France)
```
Comment (pmconrad) [inline seed-nodes.txt]: Inactive, please remove

File: seed-nodes.txt [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+"bts-seed1.abit-more.com:62015",     // abit         (China)
+"node.blckchnd.com:4243",            // blckchnd     (Germany)
+"seed.bitsharesdex.com:50696",       // iHashFury    (Europe)
+"seed.bitsharesnodes.com:1776",      // wackou       (Netherlands)
```
Comment (pmconrad) [inline seed-nodes.txt]: Inactive, please remove

File: seed-nodes.txt [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+"seed.blocktrades.us:1776",          // BlockTrades  (USA)
```
Comment (pmconrad) [inline seed-nodes.txt]: Inactive, please remove

File: seed-nodes.txt [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+"seed.cubeconnex.com:1777",          // cube         (USA)
```
Comment (pmconrad) [inline seed-nodes.txt]: Inactive, please remove

**Comment by oxarbitrage** [comment]
> Agree. I like it. :-)
Ok , i don't have anything against it apart from being a bit ugly as mentioned, the include do what is needed.

> I think you can get rid of the CMakeLists change if you #include with double quotes not angled brackets.

It will look better to me if we can do this changes, it will make the intention more clear as it will distinct it from normal angled bracket headers plus save the CMakeList changes.

File: seed-nodes.txt (hunk 1 above)
Comment (crypto-ape) [reply seed-nodes.txt]: From the C++ stand-point it is correct and it provides two benefits:

- adding next entry won't require modification of the previously last entry
- it is symmetrical :)

**Comment by crypto-ape** [comment]
I have refactored the PR.

- the CMakeLists.txt is not modified
- the raw include is used ("seed-nodes.txt")
- the added file with seed node entries also contains link to the forum topic
- the inactive nodes are removed (@pmconrad You can check this, I also inspected the servers with `nc -vc address port`.)

**Comment by pmconrad** [comment]
Hm somehow this looks wrong. I meant to remove puppies, Harvey, lafona, wackou, BlockTrades, cube.
You left in Harvey but removed liondani. liondani seems to be a bit unreliable, but please leave him in.
I always get "Connection refused" for Harvey, was it successful for you?

lafona has a new seed: `bts.lafona.net:1776`, please re-add.

**Comment by crypto-ape** [comment]
(I don't remember, but now I obtained a refusal from Harvey as You did.)

I have re-added lafona and removed Harvey.

**Comment by pmconrad** [comment]
Thanks!

**Pull request merged**